pkg/api/types.go

```diff
@@ -1510,43 +1510,64 @@ func (k *KubernetesConfig) IsAzureCNIMonitoringEnabled() bool {
 }
 
 // IsContainerMonitoringEnabled checks if the container monitoring addon is enabled
-func (k *KubernetesConfig) IsContainerMonitoringEnabled() bool {
-	return k.isAddonEnabled(ContainerMonitoringAddonName, DefaultContainerMonitoringAddonEnabled)
+func (p *Properties) IsContainerMonitoringEnabled() bool {
+	if p.OrchestratorProfile == nil || p.OrchestratorProfile.KubernetesConfig == nil || p.IsAzureStackCloud() {
+		return false
+	}
+	return p.OrchestratorProfile.KubernetesConfig.isAddonEnabled(ContainerMonitoringAddonName, DefaultContainerMonitoringAddonEnabled)
 }
 
 // IsTillerEnabled checks if the tiller addon is enabled
 func (k *KubernetesConfig) IsTillerEnabled() bool {
 	return k.isAddonEnabled(DefaultTillerAddonName, DefaultTillerAddonEnabled)
 }
 
-// IsAADPodIdentityEnabled checks if the tiller addon is enabled
-func (k *KubernetesConfig) IsAADPodIdentityEnabled() bool {
-	return k.isAddonEnabled(DefaultAADPodIdentityAddonName, DefaultAADPodIdentityAddonEnabled)
+// IsAADPodIdentityEnabled checks if the AAD pod identity addon is enabled
+func (p *Properties) IsAADPodIdentityEnabled() bool {
+	if p.OrchestratorProfile == nil || p.OrchestratorProfile.KubernetesConfig == nil || p.IsAzureStackCloud() {
```

Comment:
We repeat this line several times. It could be a helper function to keep things more DRY:

```go
func (p *Properties) isKubernetesAndNotAzureStack() bool {
    return p.OrchestratorProfile != nil && p.OrchestratorProfile.KubernetesConfig != nil && !p.IsAzureStackCloud()
}

func (p *Properties) IsACIConnectedEnabled() bool {
    return p.isKubernetesAndNotAzureStack() &&
        p.OrchestratorProfile.KubernetesConfig.isAddonEnabled(DefaultACIConnectorAddonName, DefaultACIConnectorAddonEnabled)
}
```

Maybe we could look at that for a future refactoring.

Reply:
@mboersma comment addressed here => #1261